packages/babel-plugin-transform-es2015-modules-commonjs/src/index.js

```diff
@@ -306,6 +306,15 @@ export default function () {
                       addTo(exports, id.node.name, id.node);
                       init.replaceWith(buildExportsAssignment(id.node, init.node).expression);
                       nonHoistedExportNames[id.node.name] = true;
+                    } else if (id.isObjectPattern()) {
+                      for (let i = 0; i < id.node.properties.length; i++) {
+                        const prop = id.node.properties[i];
+                        if (!t.isRestProperty(prop)) {
```

Comment:
It's fine for people to have enabled support for the syntax without enabling the actual transformation, so I think it would be good to handle.